owner-onboarding-server/src/ov_management/mod.rs

```diff
@@ -0,0 +1,363 @@
+use crate::OwnerServiceUD;
+use fdo_data_formats::{ownershipvoucher::OwnershipVoucher, types::Guid, DeserializableMany};
+use openssl::pkey::{PKey, Private};
+use serde::Serialize;
+use serde_json::json;
+use std::io::Read;
+use std::sync::Arc;
+use std::{str::FromStr, usize};
+use thiserror::Error;
+use warp::{http::HeaderValue, hyper::StatusCode, Buf};
+use warp::{Filter, Rejection};
+
+pub(crate) fn ov_filter(
+    session_store: Arc<OwnerServiceUD>,
+) -> impl Filter<Extract = impl warp::Reply, Error = warp::Rejection> + Clone {
+    let store1 = session_store.clone();
+    let baseurl = warp::path("management").and(warp::path("v1"));
+
+    let add = baseurl
+        .and(warp::post())
+        .and(warp::path("ownership_voucher"))
+        .and(warp::path::end())
+        .and(warp::header::value("X-Number-Of-Vouchers"))
+        .and(warp::header::value("content-type"))
+        .and(warp::body::content_length_limit(1024 * 32))
+        .and(warp::body::bytes())
+        .map(move |count, content_type, body| (count, content_type, body, session_store.clone()))
+        .untuple_one()
+        .and_then(add_ov);
+
+    let delete = baseurl
+        .and(warp::post())
+        .and(warp::path("ownership_voucher"))
+        .and(warp::path("delete"))
+        .and(warp::path::end())
+        .and(warp::header::exact("content-type", "application/json"))
+        .and(warp::body::content_length_limit(1024 * 32))
+        .and(warp::body::json())
+        .map(move |uuids: Vec<String>| (uuids, store1.clone()))
+        .untuple_one()
+        .and_then(delete_ov)
+        .recover(recover_err);
+
+    add.or(delete)
+}
+
+async fn add_ov(
+    count: HeaderValue,
+    content_type: HeaderValue,
+    body: impl Buf,
+    user_data: Arc<OwnerServiceUD>,
+) -> core::result::Result<Box<dyn warp::reply::Reply>, warp::Rejection> {
+    // check for ov type
+    let content_type = content_type.to_str().unwrap();
```

Comment:
generally, we cannot bail on a server, we should return an error and continue listening to requests